charmcraft/commands/store/registry.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Module to work with OCI registries."""
+
+import logging
+from urllib.request import parse_http_list, parse_keqv_list
+
+import requests
+
+from charmcraft.cmdbase import CommandError
+
+logger = logging.getLogger(__name__)
+
+# some mimetypes
+MANIFEST_LISTS = 'application/vnd.docker.distribution.manifest.list.v2+json'
```

Comment:
Googled this and found surprisingly few hits. Not a friendly search term I guess :rofl: Found some docs @ https://docs.docker.com/registry/spec/manifest-v2-2/

Reply:
yes, those are the ONLY docs...